cmd/prepare-storage.go

```diff
@@ -198,8 +198,15 @@ func connectLoadInitFormats(verboseLogging bool, firstDisk bool, endpoints Endpo
 		return nil, nil, err
 	}
 
+	// Return error when quorum unformatted disks - indicating we are
+	// waiting for first server to be online.
+	unformattedDisks := quorumUnformattedDisks(sErrs)
```

Comment:
This actually means many of new fresh setups will have many unformatted disks at the first.. healing will detect that in 10 seconds and will reformat again but it will print a lot of unnecessary logs etc.. maybe let's ignore the offline disk after 2 minutes ? (time.Since(globalBootTime) > 2 *time.Minute)

Reply:
where would I add that? @vadmeste

Reply:
maybe something like below, again the only point is to generate less logs which can be confusing for users.

```go
diff --git a/cmd/format-erasure.go b/cmd/format-erasure.go
index d535039b8..662887d10 100644
--- a/cmd/format-erasure.go
+++ b/cmd/format-erasure.go
@@ -305,8 +305,9 @@ func countErrs(errs []error, err error) int {
 }
 
 // Check if unformatted disks are equal to write quorum.
-func quorumUnformattedDisks(errs []error) bool {
-       return countErrs(errs, errUnformattedDisk) >= (len(errs)/2)+1
+func quorumUnformattedDisks(errs []error) (bool, int) {
+       unformattedFound := countErrs(errs, errUnformattedDisk)
+       return unformattedFound >= (len(errs)/2)+1, unformattedFound
 }
 
 // loadFormatErasureAll - load all format config from all input disks in parallel.
diff --git a/cmd/prepare-storage.go b/cmd/prepare-storage.go
index 0b6c5dd2f..522328fd7 100644
--- a/cmd/prepare-storage.go
+++ b/cmd/prepare-storage.go
@@ -200,13 +200,13 @@ func connectLoadInitFormats(verboseLogging bool, firstDisk bool, endpoints Endpo
 
        // Return error when quorum unformatted disks - indicating we are
        // waiting for first server to be online.
-       unformattedDisks := quorumUnformattedDisks(sErrs)
-       if unformattedDisks && !firstDisk {
+       quorumUnformatted, unformattedFound := quorumUnformattedDisks(sErrs)
+       if quorumUnformatted && !firstDisk {
                return nil, nil, errNotFirstDisk
        }
 
        // All disks report unformatted we should initialized everyone.
-       if unformattedDisks && firstDisk {
+       if quorumUnformatted && firstDisk && (unformattedFound == len(storageDisks) || time.Since(globalBootTime) > 2*time.Minute) {
                logger.Info("Formatting %s pool, %v set(s), %v drives per set.",
                        humanize.Ordinal(poolCount), setCount, setDriveCount)
 

```